sdk/data_cosmos/src/operations/replace_document.rs

```diff
@@ -65,11 +65,13 @@ impl<D: Serialize + Send + 'static> ReplaceDocumentBuilder<D> {
                 .unwrap_or_else(|| self.client.partition_key_serialized());
             add_as_partition_key_header_serialized2(partition_key, &mut request);
 
-            azure_core::headers::add_mandatory_header2(&self.indexing_directive, &mut request)?;
-            azure_core::headers::add_optional_header2(&self.if_match_condition, &mut request)?;
-            azure_core::headers::add_optional_header2(&self.if_modified_since, &mut request)?;
-            azure_core::headers::add_optional_header2(&self.consistency_level, &mut request)?;
-            azure_core::headers::add_mandatory_header2(&self.allow_tentative_writes, &mut request)?;
+            request.insert_headers(&self.indexing_directive);
+            request.insert_headers(&self.if_match_condition);
+            request.insert_headers(&self.if_modified_since);
+            if let Some(cl) = &self.consistency_level {
+                request.insert_headers(cl);
+            }
```

Comment:
Something which stands out to me here is that `insert_headers` no longer returns a `Response`. I believe this would allow us to write the following:

```rust
&self.consistency_level.map(request.insert_headers);
```

Reply:
Unfortunately, we need to do the following to get it to compile:

```rust
self.consistency_level
                .as_ref()
                .map(|c| request.insert_headers(c));
```
This doesn't really save us very much in terms of readability...

Reply:
oh no, you're right!